src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -3757,6 +3781,7 @@ void ExportMusicXml::rest(Rest* rest, int staff)
 
     QString noteTag = QString("note");
     noteTag += color2xml(rest);
+    noteTag += restPosition(this, rest);
```

Comment:
I wonder if we could simply use `notePosition(this, dynamic_cast<Note*>(rest))` here, i.e. fully reuse the existing code rather than duplicating it? Make the entire fix a one-liner.

Reply:
Or, probably much cleaner, change notePosition to this:

```
//---------------------------------------------------------
//   noteRestPosition
//---------------------------------------------------------

QString ExportMusicXml::noteRestPosition(const ExportMusicXml* const expMxml, const ChordRest* const cr)
{
    QString res;

    IF_ASSERT_FAILED(!cr->isNote() && !cr->isRest()) {
        return "";
    }

    if (configuration()->musicxmlExportLayout()) {
        const double pageHeight  = expMxml->getTenthsFromInches(expMxml->score()->styleD(Sid::pageHeight));

        double measureX = expMxml->getTenthsFromDots(cr->measure()->pagePos().x());
        double measureY = pageHeight - expMxml->getTenthsFromDots(cr->measure()->pagePos().y());
        double noteX = expMxml->getTenthsFromDots(cr->pagePos().x());
        double noteY = pageHeight - expMxml->getTenthsFromDots(cr->pagePos().y());

        res += QString(" default-x=\"%1\"").arg(QString::number(noteX - measureX, 'f', 2));
        res += QString(" default-y=\"%1\"").arg(QString::number(noteY - measureY, 'f', 2));
    }

    return res;
}

Reply:
Maybe even call it `chordRestPosition()` and drop the `IF_ASSERT_FAILED` part entirely, as it should be legal for chord too.

Reply:
Actually just forget about it, it just doesn't work. And the code isn't 100% identical either, `notePosition()` picks on the chord a note belongs to, `restPosition()`(of course) does not.
So back to your initial approach, 2 methods for 2 types, which similar but not identical code.
Sorry for the mess...